test/client/fixtures/sandbox/event/internal-listeners-test.js

```diff
@@ -611,3 +617,119 @@ if (browserUtils.isIE && browserUtils.version >= 10) {
+    // NOTE: After adding an element to a document which differs from the document where the element was created,
+    // some browsers automatically replace the element prototype's methods
+    // with methods of the element prototype from the different window.
+    var getListenersModule = function (iframeListenersModule, topLevelListenersModule) {
+        return browserUtils.isWebKit || browserUtils.isMSEdge ? topLevelListenersModule : iframeListenersModule;
+    };
+
+    iframe.id = 'test_unique_id_qrsdcz';
+    window.QUnitGlobals.waitForIframe(iframe)
+        .then(function () {
+            var iframeDocument            = iframe.contentDocument;
+            var iframeHammerhead          = iframe.contentWindow['%hammerhead%'];
+            var iframeListeners           = iframeHammerhead.get('./sandbox/event/listeners');
+            var targetListeners           = getListenersModule(iframeListeners, Listeners);
+            var storedBeforeDispatchEvent = targetListeners.beforeDispatchEvent;
+            var storedAfterDispatchEvent  = targetListeners.afterDispatchEvent;
+
+            targetListeners.beforeDispatchEvent = function (el) {
+                ok(!iframe.contentWindow[dispatchedEventFlag]);
+                ok(!window[dispatchedEventFlag]);
+
+                storedBeforeDispatchEvent(el);
+
+                ok(iframe.contentWindow[dispatchedEventFlag]);
+                ok(!window[dispatchedEventFlag]);
+            };
+            targetListeners.afterDispatchEvent = function (el) {
+                ok(iframe.contentWindow[dispatchedEventFlag]);
+                ok(!window[dispatchedEventFlag]);
+
+                storedAfterDispatchEvent(el);
+
+                ok(!iframe.contentWindow[dispatchedEventFlag]);
+                ok(!window[dispatchedEventFlag]);
+            };
+
+            iframeDocument.body.appendChild(link);
+
+            dispatchEvent(link, 'click');
+            link.click();
+
+            if (document.fireEvent)
+                link.fireEvent('click');
+
+            iframe.parentNode.removeChild(iframe);
+            start();
+        });
+    document.body.appendChild(iframe);
+});
+
+if (browserUtils.isIE && !browserUtils.isMSEdge) {
+    asyncTest('setSelection', function () {
+        var iframe              = document.createElement('iframe');
+        var testInput           = document.createElement('input');
+        // NOTE: To prevent the export of the constant and modification of the Listeners module export,
+        // we declare the constant in the test again.
+        var dispatchedEventFlag = 'hammerhead|event-sandbox-dispatch-event-flag';
+        var counter             = 0;
+
+        testInput.addEventListener('focus', function () {
+            iframe.parentNode.removeChild(iframe);
+            start();
+        });
+
+        testInput.value = 'test';
+        iframe.id = 'test_unique_id_qrsj12vbz';
```

Comment:
imho it's better to add some empty lines between blocks of code: here, [here](https://github.com/DevExpress/testcafe-hammerhead/pull/531/files#diff-b522bdf9c15ce13761bbb3a23dc6a2c9R714), [here](https://github.com/DevExpress/testcafe-hammerhead/pull/531/files#diff-b522bdf9c15ce13761bbb3a23dc6a2c9R724) etc.. It'll be more readable